rust/lanzatool/src/generation.rs

```diff
@@ -51,41 +47,45 @@ impl Generation {
             .try_into()
             .map_err(|err: &'static str| anyhow!(err))?;
 
-        let extensions = Self::extract_extensions(&bootspec)?;
-
         Ok(Self {
             version: link.version,
             specialisation_name: None,
-            spec: ExtendedBootJson {
-                bootspec,
-                extensions,
-            },
+            spec: ExtendedBootJson { bootspec },
         })
     }
 
-    fn extract_extensions(bootspec: &BootJson) -> Result<SecureBootExtension> {
-        Ok(Deserialize::deserialize(
-            bootspec.extensions.get("lanzaboote")
-            .context("Failed to extract Lanzaboote-specific extension from Bootspec, missing lanzaboote field in `extensions`")?
-            .clone()
-            .into_deserializer()
-        )?)
-    }
-
     pub fn specialise(&self, name: &SpecialisationName, bootspec: &BootJson) -> Result<Self> {
         Ok(Self {
             version: self.version,
             specialisation_name: Some(name.clone()),
             spec: ExtendedBootJson {
                 bootspec: bootspec.clone(),
-                extensions: Self::extract_extensions(bootspec)?,
             },
         })
     }
 
     pub fn is_specialized(&self) -> Option<SpecialisationName> {
         self.specialisation_name.clone()
     }
+
+    /// Describe the generation in a single line.
+    ///
+    /// This is currently implemented by poking around the filesystem to find the necessary data.
+    /// Ideally, the needed data should be included in the bootspec.
+    pub fn describe(&self) -> Result<String> {
+        let toplevel = &self.spec.bootspec.toplevel.0;
+
+        let nixos_version = fs::read_to_string(toplevel.join("nixos-version"))
+            .unwrap_or_else(|_| String::from("Unknown"));
+        let kernel_version =
+            read_kernel_version(toplevel).context("Failed to read kernel version.")?;
+        let build_time = read_build_time(toplevel).unwrap_or_else(|_| String::from("Unknown"));
+
+        Ok(format!(
+            "Generation {} NixOS {}, Linux Kernel {}, Built on {}",
```

Comment:
I guess you want to generation first for ordering? If so, that would be a good comment here.